### PR TITLE
Unify UQ interfaces under the uq module

### LINKED ISSUE #33: Provide a common interface for UQ + Surrogate across all UQ methods
We need an interface to support different types of UQ + Surrogates. We currently have 3 UQs, whith their code convoluted. We need a single entry point. This is necessary for us to provide clean support of different precisions for the surrogate implementations.

### PR DESCRIPTION
**Pull request by ggeorgakoudis**
- Separate random from hdcache

Closes #33

### CODE REVIEW

File: examples/main.cpp [hunk 1]
```diff
@@ -195,6 +195,8 @@ int run(const char *device_name,
     uq_policy = AMSUQPolicy::DeltaUQ_Max;
   else if (strcmp(uq_policy_opt, "deltauq-mean") == 0)
     uq_policy = AMSUQPolicy::DeltaUQ_Mean;
+  else if (strcmp(uq_policy_opt, "random") == 0)
+    uq_policy = AMSUQPolicy::RandomUQ;
```
Comment (koparasy) [inline examples/main.cpp:199]: I think we have a bug in the uq_policy_opt flag.It is set on a default value of "faiss-mean". What happens if the user does not specify the uq_policy_opt flag and does not provide an FAISS index. Should we just default to random?

**Review by koparasy — CHANGES_REQUESTED** [review]

**Comment by koparasy** [comment]
Does this fix #33  ? If so please tag it.

File: examples/main.cpp (hunk 1 above)
Comment (ggeorgakoudis) [reply examples/main.cpp:199]: You're right. How about we remove the default and we error check in the UQ constructor that there is always a path when specifying FAISS policies?

**Comment by ggeorgakoudis** [comment]
> Does this fix #33 ? If so please tag it.

Done

File: examples/main.cpp (hunk 1 above)
Comment (koparasy) [reply examples/main.cpp:199]: Sounds good.

Comment (ggeorgakoudis) [reply examples/main.cpp:199]: @koparasy Updates to review in added commit

**Review by koparasy — APPROVED** [review]

**Comment by ggeorgakoudis** [comment]
Merged

**Pull request merged**